go.mod

```diff
@@ -128,7 +128,7 @@ require (
 	golang.org/x/tools v0.0.0-20190920225731-5eefd052ad72
 	gonum.org/v1/gonum v0.0.0-20190331200053-3d26580ed485
 	google.golang.org/api v0.6.1-0.20190607001116-5213b8090861
-	google.golang.org/grpc v1.23.1
+	google.golang.org/grpc v1.26.0
```

Comment:
Notable things in the release notes for these releases:

* 1.26
  * dns: disable SRV record lookups unless grpclb is imported (#3149)
  * dns: stop polling for updates; use UpdateState API (#3165)
* 1.25
  * clientconn: override authority with address's ServerName, if set (#3073) Special Thanks: @jpbetz

@jpbetz, can you verify these don't impact the etcd client, and help us understand if upgrading grpc underneath the etcd client is reasonable? (as a point of reference, the current version of the etcd client we use, 3cf2f69b5738fb702ba1a935590f36b52b18979b aka v3.4.3, expects grpc v1.23.1 - https://github.com/etcd-io/etcd/blob/3cf2f69b5738fb702ba1a935590f36b52b18979b/go.mod#L46)

Reply:
"clientconn: override authority with address's ServerName" (#3073) is safe. I've run https://github.com/jpbetz/etcd/commit/aa6a86e3e86b3a9a7781ac8685b8952613ed61f5#diff-3aa2435f0785d80e2f2472b60d17f905 against etcd v3.4.3 with grpc 1.26.0 to verify.

Reply:
"dns: disable SRV record lookups unless grpclb is imported" (grpc/grpc-go#3149) looks okay as well. etcd uses it's own dns SRV lookup code (https://github.com/etcd-io/etcd/blob/master/pkg/srv/srv.go), not the grpc code. I verified etcd srv discovery still works with grpc 1.26.0 via:

```
<edit go.mod, set grpc to 1.26.0>
make build-docker-dns-srv-test
make docker-dns-srv-test-certs-run
```

Reply:
"dns: stop polling for updates; use UpdateState API" (grpc/grpc-go#3165) does not appear to cause problems for the etcd client. It already returns the appropriate error from build if there are no connections available (https://github.com/etcd-io/etcd/blob/master/clientv3/balancer/balancer.go#L89). Tested this via:

```
# edit go.mod, set grpc to 1.26.0
# run etcd server in a terminal
# edit /etc/hosts, set etcd1.home to an valiid IP of 127.0.0.1
$ bin/etcdctl --endpoints=etcd1.home:2379 put a success
# edit /etc/hosts, set etcd1.home to an invalid IP of 127.0.0.2
$ bin/etcdctl --endpoints=etcd1.home:2379 get a
{"level":"warn","ts":"2020-01-14T13:54:12.152-0800","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-73ef456f-aa89-43b9-9829-b9d75bf622af/etcd1.jpbetz.home:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest
 balancer error: all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.2:2379: connect: connection refused\""}

$ bin/etcdctl --endpoints=etcd1.jpbetz.home:2379 get a --command-timeout=5m --dial-timeout=5m
# edit /etc/hosts, set etcd1.home to an valid IP of 127.0.0.1
success
```

Reply:
Looks like the changes to grpc are okay from an etcd perspective. I was reasoned through each one briefly and constructed a test to verify the behavior remains as we expect. Please have a quick look at my methodology here with a critical eye. I'm happy to dig deeper if any sees points out any gaps in how I've tested these changes.

Reply:
^ @jingyih and @yoyinzyc let me know if you're aware of how the above test methodology might miss anything important

Reply:
> ^ Jingyi Hu and Yuchen Zhou let me know if you're aware of how the above test methodology might miss anything important

Generally looks good. Do we need to test sending request through [grpc gateway](https://github.com/etcd-io/etcd/blob/master/Documentation/dev-guide/api_grpc_gateway.md)?